DQMOffline/L1Trigger/interface/L1TTauOffline.h

```diff
@@ -0,0 +1,264 @@
+
+private:
+  void bookTauHistos(DQMStore::IBooker &);
+  /* void bookPhotonHistos(DQMStore::IBooker &); */
+
+  //other functions
+  double Distance(const reco::Candidate & c1, const reco::Candidate & c2);
+  double DistancePhi(const reco::Candidate & c1, const reco::Candidate & c2);
+  double calcDeltaPhi(double phi1, double phi2);
+
+  math::XYZPoint PVPoint_;
+
+  HLTConfigProvider m_hltConfig;
+
+  edm::ESHandle<MagneticField> m_BField;
+  edm::ESHandle<Propagator> m_propagatorAlong;
+  edm::ESHandle<Propagator> m_propagatorOpposite;
+
+  //variables from config file
+  edm::EDGetTokenT<reco::PFTauCollection> theTauCollection_;
+  edm::EDGetTokenT<reco::PFTauDiscriminator> AntiMuInputTag_;
+  edm::EDGetTokenT<reco::PFTauDiscriminator> AntiEleInputTag_;
+  edm::EDGetTokenT<reco::PFTauDiscriminator> DecayModeFindingInputTag_;
+  edm::EDGetTokenT<reco::PFTauDiscriminator> comb3TInputTag_;
+  edm::EDGetTokenT<reco::MuonCollection> MuonInputTag_;
+  edm::EDGetTokenT<reco::PFMETCollection> MetInputTag_;
+  edm::EDGetTokenT<reco::VertexCollection> VtxInputTag_;
+  edm::EDGetTokenT<reco::BeamSpot> BsInputTag_;
+  edm::EDGetTokenT<trigger::TriggerEvent> triggerEvent_;
+  std::string trigProcess_;
+  edm::EDGetTokenT<edm::TriggerResults> triggerResults_;
+  std::vector<std::string> triggerPath_;
+  std::string histFolder_;
+  std::string efficiencyFolder_;
+  edm::EDGetTokenT<l1t::TauBxCollection> stage2CaloLayer2TauToken_;
+  std::vector<double> tauEfficiencyThresholds_;
+  std::vector<double> tauEfficiencyBins_; 
+
+  std::vector<const reco::Muon*>  m_TightMuons;
+  std::vector<const reco::PFTau*>  m_ProbeTaus;
+  std::vector<TauL1TPair>  m_TauL1tPairs;
+
+
+  std::vector<reco::PFTauCollection>  m_RecoTaus;		
+  std::vector<l1t::TauBxCollection>  m_L1tTaus;		  
+  std::vector<reco::PFTau>  m_RecoRecoTaus;		
+  BXVector<l1t::Tau>  m_L1tL1tTaus;		  
+  
+  // config params
+  std::vector<int> m_L1tPtCuts;
+
+  float m_MaxTauEta;
+  float m_MaxL1tTauDR;
+  float m_MaxHltTauDR;
+
+  std::vector<int> m_trigIndices;
+
+  // Histograms
+  MonitorElement* h_nVertex_;
+  MonitorElement* h_tagAndProbeMass_;
+
+  // electron reco vs L1
+  MonitorElement* h_L1TauETvsTauET_EB_;
+  MonitorElement* h_L1TauETvsTauET_EE_;
+  MonitorElement* h_L1TauETvsTauET_EB_EE_;
+
+  MonitorElement* h_L1TauPhivsTauPhi_EB_;
+  MonitorElement* h_L1TauPhivsTauPhi_EE_;
+  MonitorElement* h_L1TauPhivsTauPhi_EB_EE_;
+
+  MonitorElement* h_L1TauEtavsTauEta_;
+
+  // electron resolutions
+  MonitorElement* h_resolutionTauET_EB_;
+  MonitorElement* h_resolutionTauET_EE_;
+  MonitorElement* h_resolutionTauET_EB_EE_;
+
+  MonitorElement* h_resolutionTauPhi_EB_;
+  MonitorElement* h_resolutionTauPhi_EE_;
+  MonitorElement* h_resolutionTauPhi_EB_EE_;
+
+  MonitorElement* h_resolutionTauEta_;
+
+  // tau turn-ons
+  std::map<double, MonitorElement*> h_efficiencyIsoTauET_EB_pass_;
```

Comment:
Double as a key? This seems like a terrible idea.
But you use it with predefined constants, so I guess it's fine.